pkgs/by-name/ca/caddy/plugins.nix

```
@@ -83,43 +84,70 @@ caddy.overrideAttrs (
 
     # xcaddy built output always uses pseudo-version number
     # we enforce user provided plugins are present and have matching tags here
-    doInstallCheck = true;
+    inherit doInstallCheck;
     installCheckPhase = ''
       runHook preInstallCheck
 
+      declare -A modules errors
       ${toShellVar "notfound" pluginsSorted}
 
-      while read kind module version; do
-        [[ "$kind" = "dep" ]] || continue
-        module="''${module}@''${version}"
-        for i in "''${!notfound[@]}"; do
-          if [[ ''${notfound[i]} = ''${module} ]]; then
-            unset 'notfound[i]'
-          fi
-        done
+      # put build info that we care about into `modules` list
+      while read -r kind module version _; do
+        case "$kind" in
+          'dep'|'=>')
+            modules["$module"]="$version"
```

```suggestion
            modules[$module]=$version
```

quoting is unnecessary both for the assignment and the subscript (a key can have spaces).

forgot to mention for subscript cases.